app/controllers/reportings_controller.rb

```diff
@@ -35,6 +35,11 @@ class ReportingsController < ApplicationController
   before_filter :find_project_by_project_id
   before_filter :authorize
 
+  before_filter :find_reporting, only: [:show, :edit, :update, :confirm_destroy, :destroy]
+  before_filter :build_reporting, only: :create
+
+  before_filter :check_visibility, except: [:create, :index, :new, :available_projects]
```

Comment:
I don't understand why you except `:create` from the visibility check. The method was called before, too. Did it turn out to be unneccessary?